### DISCUSSION TITLE
Filters and order by

### DISCUSSION THREAD

**Discussion by darren-glanville** [filamentphp/filament#6661] [Help] [answered]
I have a filter that applies a scope to it is enabled. The scope has a different order by that the normal.

How do I apply a customer ->orderBy('start_date', 'desc') if the filter is on. This is what I have so far:

`Tables\Filters\Filter::make('filters')
                    ->form([
                        // current_bookings
                        Forms\Components\Toggle::make('current_bookings')
                            ->label(__('ev-bookings.columns.current-bookings'))
                            ->default(true),
                    ])
                    ->query(function (Builder $query, array $data): Builder {
                        return $query
                            ->when(
                                $data['current_bookings'] == true,
                                fn (Builder $query): Builder => $query->currentBookings(),
                                fn (Builder $query): Builder => $query->oldBookings()
                            );
                    })`

**Comment by danharrin** [comment, marked as answer]
It's a bit long winded. You have to create a new class that extends Filter, and override `applyToBaseQuery()`. You can apply the order there, and everything else in `->query()` / by overriding `apply()`.

The reason is because the `->query()` gets grouped into a `where` to ensure that it does not interfere with other filters, but this means you can't order inside it.

**Reply by darren-glanville** [reply]
Thats great thanks for that, worked like a charm. Will be worth this going in the documentation.
